src/score_hv/stats_utils.py

```diff
@@ -7,43 +8,110 @@
-
-    try:
-        assert sumweights >= 0.999 * 4. * np.pi
-        assert sumweights <= 1.001 * 4. * np.pi
-    except AssertionError as err:
-        msg = (f'{gridcell_area_weights}\n'
-               '(gridcell area weights) sum does not equal 4pi steradians; ' 
-               'cannot calculate accurate global/regional weighted statistics')
-        raise AssertionError(msg) from err
-    return(weighted_mean)
-
-def area_weighted_variance(xarray_variable, gridcell_area_weights,
-                           expected_value=None):
-    """returns the gridcell weighted variance of the requested variables using
-    the following formula:
-                        
-        variance = sum_R{ w_i * (x_i - xbar)^2 },
-    
-    where sum_R represents the summation for each value x_i over the region of 
-    interest R with normalized gridcell area weights w_i and weighted mean xbar
+class var_stats:
     """
-    if expected_value == None:
-        expected_value = calculate_weighted_means(xarray_variable,
-                                                  gridcell_area_weights)
-    
-    weighted_variance = -expected_value**2 + np.ma.sum(
-                                               xarray_variable**2 * 
-                                               (gridcell_area_weights / 
-                                                gridcell_area_weights.sum()))
-    return(weighted_variance)
\ No newline at end of file
+      This allows for the instantiation of the var_stats class.  The 
+      requested list of statistics that the user wants is passed into this
+      method and is used in the calculate_requested_statistics method
+      below to calculate the desired statistics.
+      """
+    def __init__(self,stats_list):
+        self.weighted_averages=[]
+        self.variances=[]
+        self.minimum=[]
+        self.maximum=[]
+        self.stats=stats_list
+   
+    def calculate_requested_statistics(self,var_data,gridcell_area_weights,temporal_mean):
+        """This function calls the methods to calculate the statistics
+           requested by the user.  
+           var_data:the variable data in an array
+                    opened with xarray in the calling
+                    routine. 
+           gridcell_area_weights:the gridcell weights are from the data file
+                                 bfg_control_1536x768_20231116.nc.
+                                 Located in the data area of score-hv.
+           temporal_mean:the temporal means array that is calculated in
+                         the calling function.                      
+           Return:Nothing is returned.
+           """
+        for stat in self.stats:
+            if stat=='mean':
+               weighted_average=self.calculate_weighted_average(var_data,gridcell_area_weights,temporal_mean) 
+            if stat=='variance':
+               self.calculate_var_variance(gridcell_area_weights,temporal_mean,weighted_average) 
```

Comment:
how is calculate_var_variance supposed to get the weighted_average value if it's in a different if statement? what happens if someone only asks for variance, wouldn't it break? 

If the variance is dependent on the weighted_average, it needs to be calculated within the if statement for variance as well.